backend/de.metas.business/src/main/java/de/metas/invoice/interceptor/C_Invoice.java

```diff
@@ -294,12 +297,34 @@ private void linkInvoiceToPaymentIfNeeded(final I_C_Invoice invoice)
 		}
 	}
 
+	@VisibleForTesting
+	static boolean canAllocateOrderPaymentToInvoice(final I_C_Order order)
+	{
+		if (order == null)
+		{
+			return false;
+		}
+		if (order.getC_Payment_ID() <= 0)
+		{
+			return false;
+		}
+
+		final boolean isPrepayOrder = Services.get(IDocTypeBL.class).isPrepay(DocTypeId.ofRepoId(order.getC_DocType_ID()));
```

Comment:
not a critical request, but please extract `Services.get(...)` to a field, so that the services are all visible at the class beginning and not stashed away in the code

Reply:
done